rfcs/20250516-profiler/README.md

```diff
@@ -0,0 +1,175 @@
+# OneCCL Profiler Design Document (RFC)
```

Comment:
Feedback are welcome -> Feedback is welcome

understrand -> understand

profiing -> profiling

alos ->also

infrom -> inform

pLease note -> Please note

Reply:
Fixed them.